app/uk/gov/hmrc/gatekeeper/services/XmlService.scala

```diff
@@ -19,19 +19,33 @@ package uk.gov.hmrc.gatekeeper.services
 import uk.gov.hmrc.gatekeeper.connectors.XmlServicesConnector
 import uk.gov.hmrc.gatekeeper.models.xml.XmlOrganisation
 import uk.gov.hmrc.gatekeeper.models.{RegisteredUser, UserId}
-import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.gatekeeper.utils.XmlServicesHelper
+import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class XmlService @Inject() (xmlServicesConnector: XmlServicesConnector)(implicit val ec: ExecutionContext) extends XmlServicesHelper {
 
   def getXmlServicesForUser(user: RegisteredUser)(implicit hc: HeaderCarrier): Future[Set[String]] = {
-    xmlServicesConnector.getAllApis().map(apis => filterXmlEmailPreferences(user, apis))
+
+    def categoriesWhereUserSelectedAllForCategory(user: RegisteredUser) =
+      user.emailPreferences.interests
+      .filter(_.services.isEmpty)
+      .map(_.regime)
+    
+    def xmlApisWhereUserSelectedAllForCategory(user: RegisteredUser) = {
+      xmlServicesConnector.getApisForCategories(categoriesWhereUserSelectedAllForCategory(user))
+      .map(_.map(_.name).toSet)
+    }
+
+    for {
+      specificApis <-  xmlServicesConnector.getAllApis().map(apis => filterXmlEmailPreferences(user.emailPreferences.interests, apis).toSet)
```

Comment:
Could be clearer as 
`      apis <-  xmlServicesConnector.getAllApis()
      specificApis = filterXmlEmailPreferences(user.emailPreferences.interests, apis).toSet`

Reply:
Done